### PR TITLE
Distribute Data Prepper with a directory structure

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This change moves Data Prepper to distributing with a directory structure as called for in #305. It does not have all the changes covered there, but get the basic structure in place.

This PR:
* Adds `data-prepper-main`
* Changes `data-prepper-core` to be a regular `jar` file which does not contain dependencies
* Moved the `main` Java method into `data-prepper-main`, along with a few resources
* Copy over all Data Prepper dependencies into the `lib/` directory of the Data Prepper directory structure
* Run Data Prepper using a classpath instead of an executable jar.
* Fixes the end-to-end tests to use library structures instead of the executable Jar.
* Updates the Docker image to pull in the Linux tar.gz archive, extract it, and run the `bin/data-prepper` script
 
### Issues Resolved

Resolves much of #305, but not all of it.
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/archives/linux/data-prepper-jdk-x64.sh [hunk 1]
```diff
@@ -8,27 +8,21 @@ if [[ $# -ne 2 ]]
...
     exit 1
 fi
 
 PIPELINES_FILE_LOCATION=$1
 CONFIG_FILE_LOCATION=$2
 DATA_PREPPER_BIN=$(dirname "$(realpath "$0")")
-DATA_PREPPER_HOME="$DATA_PREPPER_BIN/.."
-EXECUTABLE_JAR=$(ls -1 "$DATA_PREPPER_HOME"/lib/*.jar 2>/dev/null)
+DATA_PREPPER_HOME=`realpath "$DATA_PREPPER_BIN/.."`
+DATA_PREPPER_CLASSPATH="$DATA_PREPPER_HOME/lib/*"
```
Comment (sbayer55) [inline release/archives/linux/data-prepper-jdk-x64.sh:20]: nit: I expected the variable to contain something like `org.opensearch.dataprepper.DataPrepperExecute` not a literal file path. Although, in context on line 34, it's clear how the variable is used.

File: release/archives/build.gradle [hunk 2]
```diff
@@ -126,8 +126,10 @@ subprojects {
 
 CopySpec archiveToTar() {
     return copySpec {
+        duplicatesStrategy = DuplicatesStrategy.EXCLUDE
```
Comment (sbayer55) [inline release/archives/build.gradle:129]: 👍

**Review by sbayer55 — APPROVED** [review]

File: release/archives/linux/data-prepper-jdk-x64.sh (hunk 1 above)
Comment (dlvenable) [reply release/archives/linux/data-prepper-jdk-x64.sh:20]: This is now a Java classpath. So it contains the directory to all the jars (we no longer have only one data-prepper-core, but all the jars are in lib/). When running Data Prepper, the script provides Java with both the classpath directories and the classname of the class with the Java `main` method.

**Review by dinujoh — COMMENTED** [review]
The changes look good. We need to update the readMe and installation guide which probably will be in future PRs.

File: release/archives/linux/data-prepper-jdk-x64.sh [hunk 3]
```diff
@@ -8,27 +8,21 @@ if [[ $# -ne 2 ]]
...
 OPENJDK=$(ls -1 $DATA_PREPPER_HOME/openjdk/ 2>/dev/null)
 
-if [[ -z "$EXECUTABLE_JAR" ]]
-then
-  echo "Jar file is missing from directory $DATA_PREPPER_HOME/lib"
-  exit 1
-fi
-
```
[the remark this answers is not in the document]
Comment (dlvenable) [reply release/archives/linux/data-prepper-jdk-x64.sh:28]: I don't think we should check for a few reasons:
* Java will fail if the `DataPrepperExecute` class is not found.
* Why only check for `data-prepper-main`? Why not `data-prepper-core`, `data-prepper-api`, etc.? Then this script becomes tightly coupled with the Jar structure.

**Comment by dlvenable** [comment]
> The changes look good. We need to update the readMe and installation guide which probably will be in future PRs.

Yes, we will update these in a future PR. I didn't want to make this one too large.

**Comment by dlvenable** [comment]
@dinujoh , Also, we have a task in #305 to update the documentation.

**Review by dinujoh — APPROVED** [review]

**Pull request merged**